ssl/statem/statem_lib.c

```diff
@@ -1195,6 +1195,7 @@ int tls_get_message_header(SSL *s, int *mt)
                      * not return success until we see the second ClientHello
                      * with a valid cookie.
                      */
+                    s->hello_retry_request = SSL_HRR_PENDING;
```

Comment:
It doesn't but since s->hello_retry_request was zapped here:
https://github.com/openssl/openssl/blob/70ed3046c5f156cab45a2bb2c8e36ab5d29a45ca/ssl/ssl_lib.c#L5840-L5842
and thus SSL_stateless will fail this test here:
https://github.com/openssl/openssl/blob/70ed3046c5f156cab45a2bb2c8e36ab5d29a45ca/ssl/ssl_lib.c#L5853-L5856
and return -1, again s_server prints "ERROR" and closes the connection.